src/main/java/delete_post/interface_adapters/DeletePostPresenter.java

```diff
@@ -12,10 +14,11 @@ public DeletePostPresenter(DeletePostViewModel viewModel){
         this.viewModel = viewModel;
     }
     @Override
-    public void prepareFailView(String error){
-        //error popup, no change to posts view or database;
+    public void prepareFailView(){
+        viewModel.updateViewModel(false, true);
     }
     public void prepareSuccessView(DeletePostResponseModel responseModel){
-        //update view posts view
+        viewModel.updateViewModel(true, true);
```

Comment:
I think you need to change the second boolean to false, otherwise it will always cause the error message to show